Add device class support for button entity (#1145)
home-assistant/developers.home-assistant: docs/core/entity/button.md

    async def async_press(self) -> None:
        """Handle the button press."""
```

### Available device classes

> Optional specifies what type of entity this. It will possibly map to Google device types.

Reviewer: ```suggestion
Optionally specifies what type of entity it is. It will possibly map to Google device types.
```

| Value | Description
| ----- | -----------
| restart | The button entity restarts the device.

> | updates | The button entity software updates the device.

Reviewer 2: ```suggestion
| updates | The button entity updates the device.
```

Reviewer 2: ```suggestion
| updates | The button entity updates the software of the device.
```

> | updates | The button entity updates the software of the device.

Author: ```suggestion
| update | The button entity updates the software of the device.
```
